prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -161,6 +165,36 @@ def canonicalize_dataset_using_columns_for_split(
             {"input_col": input_col, "output_col": output_col}
         )
 
+    @staticmethod
+    def automatic_column_selection(
+        instruction: str, dataset_name: str, dataset_columns: str, example_rows: dict
+    ):
+        """Find appropriate input and output columns for a given dataset and tasks."""
+        prompt = construct_prompt_for_column_selection(
+            instruction, dataset_name, dataset_columns, example_rows
+        )
+        required_keys = ["input", "output"]
+        optional_keys = ["ambiguous", "irrelevant"]
+
+        response = parse_prompt_to_fields(prompt, required_keys, optional_keys)
+        input_columns = response["input"]
+        output_column = response["output"]
+
+        if len(input_columns) < 1 or len(output_column) != 1:
+            raise RuntimeError(
+                "Input columns length was less than 1 or output column length was not 1"
+            )
+
+        incorrect_columns = [
+            col for col in input_columns + output_column if col not in dataset_columns
+        ]
+        if len(incorrect_columns) > 0:
+            raise RuntimeError(
+                "Incorrect columns being parsed"
+            )  # TODO: potentially could also mean that the prompt needs to be updated
```

Comment:
Made the error a bit more descriptive.
```suggestion
            raise RuntimeError(
                f"One or more columns ({incorrect_columns}) were output that were "
                f"not in the list of columns in the dataset ({dataset_columns})."
            )  # TODO: potentially could also mean that the prompt needs to be updated
```